tempesta_fw/t/functional/regression/test_health_monitor.py

```diff
@@ -0,0 +1,288 @@
+        assert m, 'Cannot parse server stats: %s\n' % (stats)
+        num = int(m.group(1))
+        assert (num == 1 and self.state) or (num == 0 and not self.state), \
+            ("Incorrect server HTTP availability state:\n"
+             "\tnum = %d, self.state = %d\n"
+             % (num, self.state))
+
+    def check_results(self):
+        result = None
+        for message in ['response', 'fwd_request']:
+            expected = getattr(self.tester.current_chain, message)
+            received = getattr(self.tester.recieved_chain, message)
+            if message == 'fwd_request' and not self.state:
+                continue
+            expected.set_expected(expected_time_delta=CHAIN_TIMEOUT)
+            if (expected != received):
+                result = (message, received, expected)
+                break
+        if not result:
+            return True
+        self.assert_msg(result[0] == 'response', result)
+        return self.tester.next_stage(result)
+
+    def assert_msg(self, cond, (message, received, expected), trans=False):
+        trans_str = ' during transition' if trans else ''
+        assert cond, \
+            ("Received message (%s) does not suit expected one%s!\n\n"
+             "\tReceieved:\n<<<<<|\n%s|>>>>>\n"
+             "\tExpected:\n<<<<<|\n%s|>>>>>\n"
+             % (message, trans_str, received, expected))
+
+
+class StagedDeproxy(deproxy.Deproxy):
+
+    def __init__(self, *args, **kwargs):
+        deproxy.Deproxy.__init__(self, *args, **kwargs)
+        self.stages = []
+        self.stages_n = 0
+        self.stages_processed = 0
+        self.current_stage = None
+
+    def run(self):
+        for _ in range(self.message_chains):
+            self.prepare()
+            self.loop()
+            if not self.check_expectations():
+                break
+
+    def prepare(self):
+        self.current_stage.prepare()
+
+    def check_expectations(self):
+        return self.current_stage.check_results()
+
+    def assert_stages(self):
+        assert self.stages_processed == self.stages_n, \
+            ("Not all stages are passed: processed"
+             " stages = %d, total stages count = %d"
+             % (self.stages_processed, self.stages_n))
+
+    def configure(self, sg_name, stages):
+        self.sg_name = sg_name
+        self.stages = stages
+        self.stages_n = len(stages)
+        self.current_stage = self.stages.pop(0)
+        self.stages_processed = 1
+
+    def next_stage(self, messages):
+        if not self.stages:
+            return False
+        self.stages[0].check_transition(messages)
+        self.current_stage = self.stages.pop(0)
+        self.stages_processed += 1
+        return True
+
+    def get_server_path(self):
+        return '%s/%s:%s' % (self.sg_name, self.servers[0].ip,
+                             self.servers[0].port)
+
+
+class TestHealthMonitor(functional.FunctionalTest):
+    """ Test for health monitor functionality with stress option.
```

Comment:
I wouldn't say it's 'regression' test. It's definitely test for a independent new feature - health monitoring. Please move the test to it's own directory. We'll need more tests about HTTP availability in future.